### DISCUSSION TITLE
1.5.4 precompiled binaries, libjpeg version

### DISCUSSION THREAD

**Discussion by catharsis71** [tjko/jpegoptim#150] [General]
Testing the Linux precompiled binary it says it's using libjpeg version 6 from 1998:

```
$ jpegoptim153 --version
jpegoptim v1.5.3  Linux (Mar 25 2023)
Copyright (C) 1996-2023, Timo Kokkonen

This program comes with ABSOLUTELY NO WARRANTY. This is free software,
and you are welcome to redistribute it under certain conditions.
See the GNU General Public License for more details.


libjpeg version: 6b  27-Mar-1998
Copyright (C) 1991-2022 The libjpeg-turbo Project and many others
```

But based on observed behavior I think they're actually using version 9?

because if I run it on a directory of files already optimized with version 8, it finds improvement on a bunch of files

if it really is using version 9 that's great because I'm looking to migrate from 8 to 9

however I have two concerns:

1.  the message on --version saying it's actually using version 6

2.  for some files (which were already optimized with version 8 in the past), it takes two passes before it's fully optimized, i.e. starting size (from version 8) = 3300547 bytes, then when I run the 1.5.4 binary on the file, it goes 3300547 --> 3191296, but then if I run it again, it goes 3191296 --> 3098050, after which it's legit finished.     Maybe this is a normal thing with version 9, haven't messed with it much yet, although it's annoying to have to do two passes to reach the end state; I can open an issue for it if you want

Pass 1:  ```./optimize1/alley/1396828854225.jpg 3264x2448 24bit N JFIF,Exif [OK] 3300547 --> 3191296 bytes (3.31%), optimized.```
Pass 2:  ```./optimize1/alley/1396828854225.jpg 3264x2448 24bit P JFIF,Exif [OK] 3191296 --> 3098050 bytes (2.92%), optimized.```
Pass 3:  ```./optimize1/alley/1396828854225.jpg 3264x2448 24bit P JFIF,Exif [OK] 3098050 --> 3098050 bytes (0.00%), skipped.```

**Comment by catharsis71** [comment]
Correction... I somehow grabbed the 1.5.3 binary instead of the 1.5.4 binary

I'm now on the 1.5.4 binary but it still shows libjpeg version 6 on --version:

```
$ jpegoptim154 --version
jpegoptim v1.5.4  Linux (May 14 2023)
Copyright (C) 1996-2023, Timo Kokkonen

This program comes with ABSOLUTELY NO WARRANTY. This is free software,
and you are welcome to redistribute it under certain conditions.
See the GNU General Public License for more details.


libjpeg version: 6b  27-Mar-1998
Copyright (C) 1991-2022 The libjpeg-turbo Project and many others
```

Testing implies that it's actually using version 9, although this time, it seems to be optimizing everything properly in a single pass (will test more to be sure)

**Reply by tjko** [reply]
Binaries are compiled against [mozjpeg](https://github.com/mozilla/mozjpeg).

"Problem" is that mozjpeg (and libjpeg-turbo) don't really differentiate themselves from the Independent JPEG Group's libjpeg. Maybe this is because they would seem to be based on (or forked from) libjpeg version 6b, and they simply never updated the version strings...
